src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -792,7 +790,7 @@ FixedwingPositionControl::set_control_mode_current(const hrt_abstime &now, bool
 		}
 
 	} else if (_control_mode.flag_control_auto_enabled && _control_mode.flag_control_climb_rate_enabled
-		   && pos_sp_curr_valid) {
+		   && _control_mode.flag_armed) {
```

Comment:
Why are we checking if it is armed? Can auto be enabled when the vehicle is disarmed?

Reply:
No this mode should never be engaged when not armed. It's purely meant as a failsafe mode.

Reply:
But then is there a need to check for the armed state? I am wondering since other modes are not checking for the armed state

Reply:
Yes because it the other modes we also want to enter if disarmed, but not the failsafe modes.
We could potentially also remove the armed check. Then the system would probably always switch into the `AUTO_ALTITUDE` or `_CLIMRATE` after boot until the `flag_control_position_enabled` is true. And we for sure would need to remove the Warnings (_Start loiter with fixed bank angle_ etc.), plus the timer that then triggers the descend. Currently they are purely set up as failsafe actions.